apis/keda/v1alpha1/scaledobject_webhook.go

```diff
@@ -297,3 +310,115 @@ func verifyCPUMemoryScalers(incomingSo *ScaledObject, action string) error {
 	}
 	return nil
 }
+
+// ValidateComplexScalingLogic validates all combinations of given arguments
+// and their values
+func ValidateComplexScalingLogic(so *ScaledObject, specs []autoscalingv2.MetricSpec) (float64, autoscalingv2.MetricTargetType, error) {
+	csl := so.Spec.Advanced.ComplexScalingLogic
+
+	if csl.Formula == "" {
+		return -1, autoscalingv2.MetricTargetType(""), fmt.Errorf("error ComplexScalingLogic.Formula needs to be specified with structure")
+	}
+
+	var num float64
+	var metricType autoscalingv2.MetricTargetType
+
+	// validate formula if not empty
+	if err := validateCSLformula(so); err != nil {
+		err := errors.Join(fmt.Errorf("error validating formula in ComplexScalingLogic"), err)
+		return -1, autoscalingv2.MetricTargetType(""), err
+	}
+	// validate target if not empty
+	num, metricType, err := validateCSLtarget(so, specs)
+	if err != nil {
+		err := errors.Join(fmt.Errorf("error validating target in ComplexScalingLogic"), err)
+		return -1, autoscalingv2.MetricTargetType(""), err
+	}
+	return num, metricType, nil
+}
+
+func validateCSLformula(so *ScaledObject) error {
+	csl := so.Spec.Advanced.ComplexScalingLogic
+
+	// if formula is empty, nothing to validate
+	if csl.Formula == "" {
+		return nil
+	}
+	// formula needs target because it's always transformed to Composite scaler
+	if csl.Target == "" {
+		return fmt.Errorf("formula is given but target is empty")
+	}
+
+	// possible TODO: this could be more soffisticated - only check for names that
+	// are used in the formula itself. This would require parsing the formula.
+	for _, trig := range so.Spec.Triggers {
+		// if resource metrics are given, skip
+		if trig.Type == cpuString || trig.Type == memoryString {
+			continue
+		}
+		if trig.Name == "" {
+			return fmt.Errorf("trigger of type '%s' has empty name but csl.Formula is defined", trig.Type)
+		}
+	}
+	return nil
```

Comment:
I think that parsing the formula and checking the naming would be a nice validation if it's something doable. Maybe using string functions... 🤔

Reply:
I could probably check all keywords that [expr language](https://github.com/antonmedv/expr) uses, save them and try to parse the names of the triggers from the formula. But maybe it would be good to set a standard for the naming of the triggers. Like using only `_` if you need a space (you cant use `-` for example, because the `expr` package evaluates that as a minus sign) and not using `.` or other symbols? 
I will try this and follow up here